src/hooks/usePersistentDominantColorFromImage.ts

```diff
@@ -0,0 +1,58 @@
+import { useEffect, useState } from 'react';
+import { PixelRatio } from 'react-native';
+import { MMKV, useMMKVString } from 'react-native-mmkv';
+import { GOOGLE_USER_CONTENT_URL } from '../components/expanded-state/unique-token/UniqueTokenExpandedStateContent';
+import { CardSize } from '../components/unique-token/CardSize';
+import { getDominantColorFromImage } from '@rainbow-me/utils';
+
+const id = 'DOMINANT_COLOR';
+enum State {
+  init,
+  loading,
+  loaded,
+  failed,
+}
+
+const storage = new MMKV({
+  id,
+});
+
+type Result = {
+  state: State;
+  result: string;
+};
+
+const size = Math.ceil(CardSize) * PixelRatio.get();
+const getLowResUrl = (url: string) => {
```

Comment:
This function is duplicated here: https://github.com/rainbow-me/rainbow/pull/2472/files#diff-ec86c9777ca223963133e263abd68388985faf046411af00fbf77ad2cc264aebR27 so it should be moved to a common place and reused